File: server/Models/Cryptid.cs
using System.ComponentModel.DataAnnotations;

namespace cryptipedia.Models;

// REVIEW cs:class code snippet

public class Cryptid : RepoItem<int>
{
  public string Name { get; set; }
  public int ThreatLevel { get; set; }
  public int Size { get; set; }
  [Url] public string ImgUrl { get; set; }
  public string Origin { get; set; }
  public string Description { get; set; }
  public string DiscovererId { get; set; }
  public Profile Discoverer { get; set; }
}

File: client/src/models/Cryptid.js
import { Profile } from "./Account.js"

export class Cryptid {
  constructor(data) {
    this.id = data.id
    this.createdAt = new Date(data.createdAt)
    this.updatedAt = new Date(data.updatedAt)
    this.name = data.name
    this.description = data.description
    this.threatLevel = data.threatLevel
    this.size = data.size
    this.origin = data.origin
    this.discoverer = new Profile(data.discoverer)
    this.discovererId = data.discovererId
    this.imgUrl = data.imgUrl
  }
}